### PR TITLE
Feat/pnl

### PR DESCRIPTION
**Pull request by cristiangpr**
Ticket: https://app.clickup.com/t/2w2x5q2

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -311,3 +319,15 @@ type FpmmPoolMembership @entity {
   "Amount of liquidity tokens owned by funder"
   amount: BigInt!
 }
+type MarketProfitPerAccount @entity {
+  "account address + market address"
```
Comment (dylangolow) [inline schema.graphql:323]: Align this with the rest and rewrite as 
"ID as account address and market maker address"

File: schema.graphql [hunk 2]
```diff
@@ -47,6 +47,10 @@ type Account @entity {
   collateralVolume: BigInt!
   "Total number of trades performed by this user"
   numTrades: BigInt!
+  "Realized profit from redemptions"
+  profit: BigDecimal!
+  "Profits by market"
+  fpmmProfits: [MarketProfitPerAccount!] @derviedFrom(field: "account")
```
Comment (dylangolow) [inline schema.graphql:53]: typo in @derivedFrom

File: schema.graphql [hunk 3]
```diff
@@ -1,313 +1,337 @@
...
+    id: ID!
+    "Timestamp at which account first interacted with Polymarket"
+    creationTimestamp: BigInt!
+    "Timestamp at which account most recently interacted with Polymarket"
+    lastSeenTimestamp: BigInt!
+    "Total volume of this user's trades in USDC base units"
+    collateralVolume: BigInt!
+    "Total number of trades performed by this user"
+    numTrades: BigInt!
+    "Realized profit in USDC from merges, sells, and for positions in resolved markets"
```
Comment (L-Kov) [inline schema.graphql:50]: any idea why this is not syntactically highlighted correctly here? seems to be the comma, I assume it's ok though.

Comment (dylangolow) [reply schema.graphql:50]: @L-Kov Must be that, looks fine in local IDE

File: src/ConditionalTokensMapping.ts [hunk 4]
```diff
@@ -192,6 +200,49 @@ export function handleConditionResolution(event: ConditionResolution): void {
...
+          let position = MarketPosition.load(pos);
+          if (position) {
+            let numerator = payoutNumerators[position.outcomeIndex.toI32()];
+            let redemptionValue = position.netQuantity
+              .times(numerator)
+              .div(payoutDenominator);
+            let averageRedemptionPrice = redemptionValue.div(
+              position.netQuantity,
+            );
+            let averagePricePaid = position.netValue.div(position.netQuantity);
```
Comment (L-Kov) [inline src/ConditionalTokensMapping.ts:227]: netValue is set when you buy?

Comment (dylangolow) [reply src/ConditionalTokensMapping.ts:227]: netValue is the total amount the user has paid to get into the current position, unique per account-market-outcome

It gets increased on buys and decreased on sells.

Comment (dylangolow) [reply src/ConditionalTokensMapping.ts:227]: So you continue to impact netValue and netQuantity on subsequent buys, sells, merges, splits

**Review by L-Kov — APPROVED** [review]

**Pull request merged**